### PR TITLE
LPS-159537 Add notification queue entry status attribute to DTO

### PR DESCRIPTION
**Pull request by mateussgn**
Related issues:
https://issues.liferay.com/browse/LPS-159537
https://issues.liferay.com/browse/LPS-159538

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
@mateussgn, could you please check if this is related to your changes? Thx.

**Comment by mateussgn** [comment]
Sure! I'll check.

**Comment by mateussgn** [comment]
@rodrigopaulino, It was a wrong class call, solved!

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/service/impl/NotificationQueueEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -171,10 +173,18 @@ public NotificationQueueEntry resendNotificationQueueEntry(
 
 	@Override
 	public void sendNotificationQueueEntries() throws PortalException {
+		List<NotificationQueueEntry> notificationQueueEntries =
+			notificationQueueEntryPersistence.findBySent(false);
```
Comment (rodrigopaulino) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/service/impl/NotificationQueueEntryLocalServiceImpl.java:177]: Put this in an `else` clause. This way two searches won't be executed at times.

File: modules/apps/notification/notification-web/src/main/java/com/liferay/notification/web/internal/frontend/data/set/view/table/NotificationQueueEntryTableFDSView.java [hunk 2]
```diff
@@ -40,6 +42,22 @@ public FDSTableSchema getFDSTableSchema(Locale locale) {
...
+			).add(
+				"fromName", "from"
+			).add(
+				"toName", "to"
+			).add(
+				"status", "status",
+				fdsTableSchemaField -> fdsTableSchemaField.setContentRenderer(
+					"notificationQueueEntryStatusDataRenderer")
+			).build();
+		}
```
Comment (rodrigopaulino) [inline modules/apps/notification/notification-web/src/main/java/com/liferay/notification/web/internal/frontend/data/set/view/table/NotificationQueueEntryTableFDSView.java:59]: You don't need to wrap this entire thing with the feature flag. You can do something like this:
```suggestion
               fdsTableSchemaBuilder.add(
				"subject", "subject"
			).add(
				"triggerBy", "trigger-by"
			).add(
				"fromName", "from"
			).add(
				"toName", "to"
			);
			
		if (!GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-159052"))) {
			fdsTableSchemaBuilder.add(
				"status", "status",
				fdsTableSchemaField -> fdsTableSchemaField.setContentRenderer(
					"notificationQueueEntryStatusDataRenderer")
			);
		}
		else {
		        fdsTableSchemaBuilder.add(
				"sent", "status",
			         fdsTableSchemaField -> fdsTableSchemaField.setContentRenderer(
				        "notificationQueueEntryStatusDataRenderer");
			);
		}
		
		return fdsTableSchemaBuilder.build();
```

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Nice work, @mateussgn. Could you please address the points below? Thx!

**Comment by rodrigopaulino** [comment]
ci:stop

**Comment by mateussgn** [comment]
Great observation, @rodrigopaulino!
Thanks for the tip!

**Comment by mateussgn** [comment]
ci:test:sf

**Review by rodrigopaulino — APPROVED** [review]
LGTM! 🎉 
Thanks, @mateussgn

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by rodrigopaulino** [comment]
ci:forward

**Pull request closed without merging**